shared-helpers/src/preferences.tsx

```diff
@@ -0,0 +1,312 @@
+import * as React from "react"
```

Comment:
No additions, this is just 80% of the previous `preferences` file from ui-components